### DISCUSSION TITLE
Upgrading client to 6.x GetDocumentsCommand wants DocumentConventions

### DISCUSSION THREAD

**Discussion by ryanheath** [ravendb/ravendb#18413] [General]
We want to upgrade the client to 6.0.101
but GetDocumentsCommand ctor wants a DocumentConventions now.

What's it for, the 6.0 documentation does not list this new parameter.
There are Default instances of DocumentConventions but they are internal only.

// Ryan

**Comment by ppekrol** [comment]
Hi @ryanheath 

This is used to determine if HTTP compression (UseHttpCompression) should be used at all and which HTTP compression algorithm (HttpCompressionAlgorithm) should be used over the wire.

You can get instance of DocumentConventions from either DocumentStore.Conventions (global conventions) or since you are asking for a command, then you probably are operating on RequestExecutor, so there is a field there - Conventions (those are global conventions with applied client configuration for that database).

I hope it helps.

**Comment by ryanheath** [comment]
Hey @ppekrol, 

I got it working (compiling) with

```
var _conventions_ = session.Advanced.RequestExecutor.Conventions;
var _command_ = new GetDocumentsCommand(_conventions_, ids, null, metadataOnly: true);
await session.Advanced.RequestExecutor.ExecuteAsync(_command_, session.Advanced.Context);
```

To be honest, it feels like something that could have been handled internally in `RequestExecutor`.

Thanks!

// Ryan
